From: Tanu Kaskinen <tanuk@iki.fi>
Date: Mon, 23 Sep 2019 07:48:48 +0300
Subject: daemon-conf: disable flat-volumes by default

Almost all distributions patch the configuration to disable
flat-volumes, because users tend to find the concept confusing (and it
also causes nasty surprises when some application pushes the volume to
100%). Let's remove the need for patching and disable the feature by
default.

Fixes: https://gitlab.freedesktop.org/pulseaudio/pulseaudio/issues/691
(cherry picked from commit ba73faa4c05717d74fbfd17e8175731e60d8841d)
---
 man/pulse-daemon.conf.5.xml.in | 2 +-
 src/daemon/daemon-conf.c       | 2 +-
 src/daemon/daemon.conf.in      | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/man/pulse-daemon.conf.5.xml.in b/man/pulse-daemon.conf.5.xml.in
index f49fa9b..1436126 100644
--- a/man/pulse-daemon.conf.5.xml.in
+++ b/man/pulse-daemon.conf.5.xml.in
@@ -237,7 +237,7 @@ License along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
       <p><opt>flat-volumes=</opt> Enable 'flat' volumes, i.e. where
       possible let the sink volume equal the maximum of the volumes of
       the inputs connected to it. Takes a boolean argument, defaults
-      to <opt>yes</opt>.</p>
+      to <opt>no</opt>.</p>
     </option>
 
   </section>
diff --git a/src/daemon/daemon-conf.c b/src/daemon/daemon-conf.c
index 33cf7a7..e74b4ba 100644
--- a/src/daemon/daemon-conf.c
+++ b/src/daemon/daemon-conf.c
@@ -68,7 +68,7 @@ static const pa_daemon_conf default_conf = {
     .realtime_priority = 5,  /* Half of JACK's default rtprio */
     .disallow_module_loading = false,
     .disallow_exit = false,
-    .flat_volumes = true,
+    .flat_volumes = false,
     .exit_idle_time = 20,
     .scache_idle_time = 20,
     .script_commands = NULL,
diff --git a/src/daemon/daemon.conf.in b/src/daemon/daemon.conf.in
index a955523..480608e 100644
--- a/src/daemon/daemon.conf.in
+++ b/src/daemon/daemon.conf.in
@@ -60,7 +60,7 @@ ifelse(@HAVE_DBUS@, 1, [dnl
 ; enable-lfe-remixing = no
 ; lfe-crossover-freq = 0
 
-; flat-volumes = yes
+; flat-volumes = no
 
 ifelse(@HAVE_SYS_RESOURCE_H@, 1, [dnl
 ; rlimit-fsize = -1
